src/components/tiles/table/use-content-change-handlers.ts

```diff
@@ -124,6 +124,10 @@ export const useContentChangeHandlers = ({
     if (!readOnly && consumerTile) {
       const sharedModelManager = consumerTile.tileEnv?.sharedModelManager;
       if (sharedModelManager?.isReady) {
+        const existingSharedTable = sharedModelManager?.findFirstSharedModelByType(SharedDataSet, tileInfo.id);
+        if (existingSharedTable) {
+          sharedModelManager?.removeTileSharedModel(consumerTile, existingSharedTable);
+        }
```

Comment:
This will sever any existing connection between the target tile and its first shared model. That's not what it looks like if you just check `sharedModelManager.findFirstSharedModelByType`, which implies that the second parameter is a `providerId`. But if you dig deeper, you'll see that `BaseDocumentContent.getFirstSharedModelByType` doesn't care if the `tileId` passed to it is the provider or not, it just returns the first shared model of the appropriate type related to the tile with the given id. So `sharedModelManager.findFirstSharedModelByType` needs to be changed so it only finds a shared model with the given `providerId` (in which case it should probably be renamed), or we should change its second parameter from `providerId` to `tileId`. I believe it was originally intended to find any `sharedModel` associated with the given `tileId` and I just misunderstood the situation when I modified the function almost a year ago.

But all this gets to a bigger question, which is how do we decide which existing connections to sever? This change prevents the geometry tile from being connected to multiple table tiles, which probably isn't desired. Will there ever be a case when we want a table tile, or datacard tile, to be connected to multiple shared data sets? Would we want a datacard tile with existing data that then gets connected to a table tile to merge content from both sources?

Reply:
Good catch -- thx for digging into this.

Reply:
Several important questions here. It is desireable for the 'target' of the shared data (the receiver? the display?) to be able to display several datasets. This is most common in graphs (draw multiple trraces on the same axes) but could theoretically be a way to merge multiple sets of data in a table or deck without changing the original source material. So we should definitely not prevent it (it works today in geometry tiles). We'd likely need new rules to display correctly in tables, and I'm not asking for that right now but it's a good idea not to prevent it.

Reply:
For tiles that can only currently display one dataset, like the table or datacard, I think we should prevent this. It will be confusing to users (and result in interruptions for developers) if we let people add multiple datasets to tiles that don't support that.

Regarding the function above, I'd vote to change the parameter name from `providerId` to just `tileId` and leave its implementation as is.

I think we have some code somewhere else that checks if the tile type is "Geometry" and then we allow multiple connections. We could do that here again. In the future we suggested adding something to the tile registration so a tile could say it supports multiple datasets. This way we don't have to have hard coded references to "Geometry" in the core CLUE code.

And based on Leslie's comment on slack I think the code is right for tiles that only support one dataset, the tileInfo tile should have its dataset severed.

Reply:
@kswenson @lbondaryk @scytacki @tealefristoe Thanks again for all the feedback. I believe the latest changes address the issues you mentioned.

As part of the updates, I went ahead and added a `supportsMultipleDataSets` option for tiles and set it to `true` for Geometry tiles only. That gets used in the check to determine if a tile should be unlinked from any tiles it's currently linked to before being linked to another.

Does that all look OK? It does seem to work correctly in the browser. In case you want to check that, here's a branch URL you can use: https://collaborative-learning.concord.org/branch/tables-can-link-to-other-tables-data-decks/?appMode=dev&unit=example

Note that I've added a few comments to the code here to explain some details. 

Please let me know what you think. Thanks.